src/matlab2tikz.m

```diff
@@ -1915,22 +1947,19 @@ function assertRegularAxes(handle)
     matlabDefaultLineWidth = 0.5;
     if m2t.args.strict ...
             || ~abs(lineWidth-matlabDefaultLineWidth) <= m2t.tol
-        if m2t.args.semanticalLineWidth
-            semStrID = find(lineWidth == m2t.args.semanticalValueList);
-            if isempty(semStrID)
-                lineOpts = opts_add(lineOpts, 'line width', sprintf('%.1fpt', lineWidth));
-            else
-                lineOpts = opts_add(lineOpts, m2t.args.semanticalStringList{semStrID}, []);
-            end
+        if m2t.args.semanticLineWidth ...
+                && ismember(lineWidth, [m2t.args.semanticLineWidthDefinition{:,2}])
```

Comment:
Could you put the second check into a separate if clause and add an else clause with a warning that there is no matching semantic width option